lib/Reader/readerir.cpp

```diff
@@ -7499,3 +7618,395 @@ void VerificationState::print() {
+          IRNode *Vector2 = Args[0];
+          ExtractedArgs.push_back((IRNode *)LLVMBuilder->CreateExtractElement(Vector2, Index0));
+          ExtractedArgs.push_back((IRNode *)LLVMBuilder->CreateExtractElement(Vector2, Index1));
+          ExtractedArgs.push_back(Args[1]);
+        }
+        break;
+
+      case 4:
+        if (Args.size() == 2) {
+          Types.push_back(Vector3Ty);
+          Types.push_back(FloatTy);
+          if (checkVectorSignature(Args, Types)) {
+            IRNode *Vector3 = Args[0];
+            ExtractedArgs.push_back((IRNode *)LLVMBuilder->CreateExtractElement(Vector3, Index0));
+            ExtractedArgs.push_back((IRNode *)LLVMBuilder->CreateExtractElement(Vector3, Index1));
+            ExtractedArgs.push_back((IRNode *)LLVMBuilder->CreateExtractElement(Vector3, Index2));
+            ExtractedArgs.push_back(Args[1]);
+          }
+        }
+        else {
+          assert(Args.size() == 3);
+          Types.push_back(Vector2Ty);
+          Types.push_back(FloatTy);
+          Types.push_back(FloatTy);
+          if (checkVectorSignature(Args, Types)) {
+            IRNode *Vector2 = Args[0];
+            ExtractedArgs.push_back((IRNode *)LLVMBuilder->CreateExtractElement(Vector2, Index0));
+            ExtractedArgs.push_back((IRNode *)LLVMBuilder->CreateExtractElement(Vector2, Index1));
+            ExtractedArgs.push_back(Args[1]);
+            ExtractedArgs.push_back(Args[2]);
+          }
+        }
+        break;
+
+      default:
+        assert(UNREACHED);
+      }
+      if (ExtractedArgs.size()) {
+        for (int i = 0; i < VectorSize; ++i) {
+          Vector = (IRNode *)LLVMBuilder->CreateInsertElement(Vector, ExtractedArgs[i], i);
+        }
+        Return = Vector;
+      }
+    }
+  }
+  if (Return) {
+    if (This) {
+      return (IRNode *)LLVMBuilder->CreateStore(Return, This); // TODO sandreenko: is volatile?
+    }
+    else {
+      return Return;
+    }
+  }
+  return 0;
+}
+
+bool GenIR::checkVectorType(IRNode *Arg) {
+  assert(Arg);
+  return Arg->getType()->isVectorTy();
+}
+
+llvm::Type *GenIR::getBaseTypeAndSizeOfSIMDType(CORINFO_CLASS_HANDLE Class, int &VectorLength, bool &IsGeneric) {
+  VectorLength = 0;
+  IsGeneric = false; 
+  static CORINFO_CLASS_HANDLE SIMDFloatHandle = 0;
+  static CORINFO_CLASS_HANDLE SIMDDoubleHandle = 0;
+  static CORINFO_CLASS_HANDLE SIMDIntHandle = 0;
+  static CORINFO_CLASS_HANDLE SIMDUShortHandle = 0;
+  static CORINFO_CLASS_HANDLE SIMDUByteHandle = 0;
+  static CORINFO_CLASS_HANDLE SIMDShortHandle = 0;
+  static CORINFO_CLASS_HANDLE SIMDByteHandle = 0;
+  static CORINFO_CLASS_HANDLE SIMDLongHandle = 0;
+  static CORINFO_CLASS_HANDLE SIMDUIntHandle = 0;
+  static CORINFO_CLASS_HANDLE SIMDULongHandle = 0;
+  static CORINFO_CLASS_HANDLE SIMDVector2Handle = 0;
+  static CORINFO_CLASS_HANDLE SIMDVector3Handle = 0;
+  static CORINFO_CLASS_HANDLE SIMDVector4Handle = 0;
+  static CORINFO_CLASS_HANDLE SIMDVectorHandle = 0;
```

Comment:
We always have a slow path for a search, so I think it is not problem. Issue  #720.